src/main/java/teammates/common/datatransfer/attributes/FeedbackQuestionAttributes.java

```diff
@@ -322,139 +323,37 @@ public int compareTo(FeedbackQuestionAttributes o) {
         }
-
-        if (getClass() != obj.getClass()) {
+        if (this.getClass() != obj.getClass()) {
             return false;
         }
 
         FeedbackQuestionAttributes other = (FeedbackQuestionAttributes) obj;
 
-        if (courseId == null) {
-            if (other.courseId != null) {
-                return false;
-            }
-        } else if (!courseId.equals(other.courseId)) {
-            return false;
-        }
-
-        if (creatorEmail == null) {
-            if (other.creatorEmail != null) {
-                return false;
-            }
-        } else if (!creatorEmail.equals(other.creatorEmail)) {
-            return false;
-        }
-
-        if (feedbackSessionName == null) {
-            if (other.feedbackSessionName != null) {
-                return false;
-            }
-        } else if (!feedbackSessionName.equals(other.feedbackSessionName)) {
-            return false;
-        }
-
-        if (giverType != other.giverType) {
-            return false;
-        }
-
-        if (numberOfEntitiesToGiveFeedbackTo != other.numberOfEntitiesToGiveFeedbackTo) {
-            return false;
-        }
-
-        if (questionNumber != other.questionNumber) {
-            return false;
-        }
-
-        if (questionMetaData == null) {
-            if (other.questionMetaData != null) {
-                return false;
-            }
-        } else if (!questionMetaData.equals(other.questionMetaData)) {
-            return false;
-        }
-
-        if (questionDescription == null) {
-            if (other.questionDescription != null) {
-                return false;
-            }
-        } else if (!questionDescription.equals(other.questionDescription)) {
-            return false;
-        }
-
-        if (questionType != other.questionType) {
-            return false;
-        }
-
-        if (recipientType != other.recipientType) {
-            return false;
-        }
-
-        if (showGiverNameTo == null) {
-            if (other.showGiverNameTo != null) {
-                return false;
-            }
-        } else if (!showGiverNameTo.equals(other.showGiverNameTo)) {
-            return false;
-        }
-
-        if (showRecipientNameTo == null) {
-            if (other.showRecipientNameTo != null) {
-                return false;
-            }
-        } else if (!showRecipientNameTo.equals(other.showRecipientNameTo)) {
-            return false;
-        }
-
-        if (showResponsesTo == null) {
-            if (other.showResponsesTo != null) {
-                return false;
-            }
-        } else if (!showResponsesTo.equals(other.showResponsesTo)) {
-            return false;
-        }
-
-        return true;
+        return Objects.equals(this.courseId, other.courseId) && Objects.equals(this.creatorEmail, other.creatorEmail)
+                && Objects.equals(this.feedbackSessionName, other.feedbackSessionName)
+                && this.giverType == other.giverType
```

Comment:
Why do some use `Object.equals` and some use `==`?

Reply:
Because enum constants are always the same object and hence `equals` is not required.